src/netops.c

```diff
@@ -171,25 +171,41 @@ int gitno_connection_data_handle_redirect(
 
 	/* Remove the service suffix if it was given to us */
 	if (service_suffix) {
+		/*
+		 * Some servers strip the query parameters from the new Location header
+		 * when sending a redirect. Others leave it in place.
+		 * Check for both, starting with the stripped case first,
+		 * since it appears to be more common.
+		 */
 		const char *service_query = strchr(service_suffix, '?');
+		size_t full_suffix_len = strlen(service_suffix);
 		size_t suffix_len = service_query ?
-			(size_t)(service_query - service_suffix) : strlen(service_suffix);
+			(size_t)(service_query - service_suffix) : full_suffix_len;
 		size_t path_len = strlen(url->path);
+		ssize_t truncate = -1;
 
 		if (suffix_len && path_len >= suffix_len) {
 			size_t suffix_offset = path_len - suffix_len;
 
 			if (git__strncmp(url->path + suffix_offset, service_suffix, suffix_len) == 0 &&
 			    (!service_query || git__strcmp(url->query, service_query + 1) == 0)) {
-				/* Ensure we leave a minimum of '/' as the path */
-				if (suffix_offset == 0)
-					suffix_offset++;
+				truncate = suffix_offset;
+			}
+		}
 
-				url->path[suffix_offset] = '\0';
+		/* Check for retained query parameters. */
+		if ((truncate == -1) && (git__suffixcmp(url->path, service_suffix) == 0)) {
```

Comment:
We usually have no braces around comparisons

Reply:
Done